design/kanister-kopia-integration.md

```diff
@@ -0,0 +1,98 @@
+# Integrating Kopia with Kanister
+
+This document proposes all the high-level changes required within Kanister to use [Kopia](https://kopia.io/) as the primary backup and restore tool.
+
+## Problems
+
+Kanister offers an in-house capability to perform backup and restore using some operation-specific Functions like BackupData, RestoreData, etc.
+Although, they are much easier to use and significantly improve the user experience of Blueprint authors, these Functions have a lot of issues.
+
+Few of those problems are;
+1. They do not encrypt in-flight data during transfer
+2. The backup data at-rest on the cloud stores are not secure
+3. Operations are slow compared to other tools in the market
+
+Therefore, they are not ideally recommended to be used for protecting application data.
+
+Instead, we make use of v2 Blueprints whose phases call a `kando` utility to perform data snapshots.
+This tool internally executes `Kopia` commands to manage backups in the cloud.
+
+This approach is again not a robust solution because;
+1. We have to create a separate kopia profile and use it within kando command to pass necessary kopia server info
+2. We have to follow the correct kando command syntax in the Blueprint while backing up and mention `--output-name` in the Action outputArtifacts
+3. Kopia errors are not logged when kando commands fail in the acion phase  
+
+## Introducing Kopia
+
+Kopia is a powerful, cross-platform tool for managing encrypted backups in the cloud.
+It provides fast and secure backups, using compression, data deduplication and client-side end-to-end encryption.
+It supports a variety of backup storage targets, including object stores, which allows users to choose the storage provider that better addresses their needs.
+It is a lock-free system that allows for concurrent multi-client operations including garbage collection.
+
+To explore other features of Kopia, see its [documentation](https://kopia.io/docs/features/).
+
+## Scope
+
+1. Design the usage of [Kopia Repository Server](https://kopia.io/docs/repository-server/) as an separate on-demand workload
+   - In order to make use of user access control and cloud storage credential abstraction offered in the server-based operations 
+2. Re-work Kanister Functions like BackupData, RestoreData, CopyVolumeData, etc. with  Kopia workflows
+   - And leverage the use of Kopia Repository Server workload within these Functions
+
+### Design the usage of Kopia Repository Server
+
+
```

Comment:
@pavannd1 Might need your help here to briefly state our CR/CRD design approach.